hbase-oss/src/main/java/org/apache/hadoop/hbase/oss/sync/TreeLockManager.java

```diff
@@ -488,7 +488,18 @@ public void close() throws IOException {
         try {
           writeUnlock(src);
         } finally {
-          writeUnlock(dst);
+          try {
+            writeUnlock(dst);
+          } finally {
+            // Tricky... HBossContract tests tough things like
+            //   `rename("/", "/somethingelse")`
+            // This means we grabbed write locks on
+            //    /               (src)
+            //    /somethingelse  (dst)
+            // Thus, we can't safely delete the znodes for src as it may
+            // then also affect the (held) lock on the dst.
+            recursiveDelete(src);
```

Comment:
One thing I'm worried about after looking at this (for the third time): is it safe to delete the znodes on the src if the rename was unsuccessful? I need to think about this.

Reply:
I think it's ok to delete it. The only con is that anyone trying to access it again would then recreate the znode in writeLock()/readLock().